base/applications/cmdutils/tasklist/tasklist.c

```diff
@@ -0,0 +1,359 @@
+// MaxWidth is the width for printing.
+// StartingUnit is the minimum memory unit used, 0 for Byte, 1 for KB, 2 for MB ...
+// return FALSE when failed to format.
+BOOL PrintMemory(SIZE_T MemorySizeByte, INT MaxWidth, INT StartingUnit)
+{
+    WCHAR *MemoryUnit[] = {L"B", L"K", L"M", L"G", L"T", L"P"};
+
+    MaxWidth -= 2; // a space and a unit will occupy 2 length.
+    if (MaxWidth <= 0)
+    {
+        return FALSE;
+    }
+
+    SIZE_T MemorySize = MemorySizeByte;
+
+    for (INT i = 0; i < _countof(MemoryUnit); i++)
+    {
+        if (i >= StartingUnit)
+        {
+            // calculate the length needed to print.
+            INT PrintLength = 0;
+            SIZE_T Tmp = MemorySize;
+            do
+            {
+                Tmp /= 10;
+                PrintLength += 1;
+            } while (Tmp);
+
+            if (PrintLength + (PrintLength - 1) / 3 <= MaxWidth) // (PrintLength - 1) / 3 is the length comma will take.
+            {
+                // enough to hold
+
+                // print padding space
+                PrintSpace(MaxWidth - (PrintLength + (PrintLength - 1) / 3));
+
+                INT Mod = 1;
+                for (INT i = 0; i < PrintLength - 1; i++)
+                {
+                    Mod *= 10;
+                }
+
+                for (INT i = PrintLength - 1; i >= 0; i--)
+                {
+                    ConPrintf(StdOut, L"%d", MemorySize / Mod);
+                    MemorySize %= Mod;
+                    if (i && i % 3 == 0)
+                    {
+                        ConPrintf(StdOut, L",");
+                    }
+                    Mod /= 10;
+                }
+
+                // print the unit.
+                ConPrintf(StdOut, L" %ls", MemoryUnit[i]);
+
+                return TRUE;
+            }
+        }
+        
+        MemorySize >>= 10; // MemorySize /= 1024;
+    }
+
+    // out of MemoryUnit range.
+    return FALSE;
+}
+
+VOID PrintHeader(VOID)
+{
+    HINSTANCE hInstance = GetModuleHandleW(NULL);
+    PrintResString(hInstance, IDS_HEADER_IMAGENAME, COLUMNWIDTH_IMAGENAME, TRUE);
+    PrintSpace(1);
+    PrintResString(hInstance, IDS_HEADER_PID,       COLUMNWIDTH_PID,       FALSE);
+    PrintSpace(1);
+    PrintResString(hInstance, IDS_HEADER_SESSION,   COLUMNWIDTH_SESSION,   FALSE);
+    PrintSpace(1);
+    PrintResString(hInstance, IDS_HEADER_MEMUSAGE,  COLUMNWIDTH_MEMUSAGE,  FALSE);
+
+    ConPrintf(StdOut, L"\n");
+
+    PrintSplitLine(COLUMNWIDTH_IMAGENAME);
+    PrintSpace(1);
+    PrintSplitLine(COLUMNWIDTH_PID);
+    PrintSpace(1);
+    PrintSplitLine(COLUMNWIDTH_SESSION);
+    PrintSpace(1);
+    PrintSplitLine(COLUMNWIDTH_MEMUSAGE);
+
+    ConPrintf(StdOut, L"\n");
+}
+
+BOOL EnumProcessAndPrint(BOOL bNoHeader)
+{
+    // Call NtQuerySystemInformation for the process information
+    ULONG ProcessInfoBufferLength = 0;
+    ULONG ResultLength = 0;
+    PBYTE ProcessInfoBuffer = 0;
```

Comment:
```suggestion
    PBYTE ProcessInfoBuffer = NULL;
```

Reply:
Please implement this suggestion as well.